FrontEnd/src/App.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import './index.css';
+import Recaptcha from 'react-recaptcha';
+
+export default class App extends Component {
+  render() {
+    return (
+      <Register />
+    );
+  }
+}
+
+// eslint-disable-next-line
+const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+
+const validateForm = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach(
+      (val) => val.length > 0 && (valid = false)
+    );    
+    return valid;
+}
+
+class Register extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.recaptchaLoaded = this.recaptchaLoaded.bind(this);
+    this.verifyCallback = this.verifyCallback.bind(this);
+    
+    this.state = {
+      firstName: null,
+      email: null,
+      message: null,
+      isVerified: false,
+      errors: {
+        firstName: '',
+        lastName: '',
+        email: '',
+        message: ''
+      }
+    };
+  }
+
+  handleChange = (event) => {
+    event.preventDefault();
+    const { name, value } = event.target;
+    let errors = this.state.errors;
+
+    switch (name) {
+      case 'firstName': 
+        errors.firstName = value.length == 0 ? 'Please enter your first name' : '';
+        break;
+
+        case 'lastName': 
+        errors.lastName = value.length == 0  ? 'Please enter your last name'  : '';
+        break;
+      case 'email': 
+        errors.email = validEmailRegex.test(value) ? '' : 'Please enter a valid email address';
+        break;
+      case 'message': 
+        errors.message = value.length == 0 ? 'Please enter your message or question' : '';
+        break;
+      default:
+        break;
+    }
+
+    this.setState({errors, [name]: value});
+  }
+ 
+  recaptchaLoaded(){
+    console.log('recaptcha loaded');
+  }
+
+  verifyCallback = (response) => {
+    console.log("verifyCallback called and set to true");
+
+    if(response){      
+      this.setState({
+        isVerified: true
+      })
+      
+    }
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();    
+
+    if(!this.state.isVerified){
+      alert('Pleave verify that you are a human using the reCaptcha'); 
+      return;
+    }
+
+    if(validateForm(this.state.errors)  && 
+            this.state.firstName != null && 
+            this.state.lastName != null && 
+            this.state.email != null && 
+            this.state.message != null){    
```

Comment:
Seems like this logic could be included in your validateForm function, no?

Reply:
Yes, there is no explanation for this one; not sure why I put it there.

Reply:
No biggie - just recognizing the logic probably belongs somewhere else.